ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/basefee-input.test.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { fireEvent, screen } from '@testing-library/react';
+
+import mockEstimates from '../../../../../test/data/mock-estimates.json';
+import mockState from '../../../../../test/data/mock-state.json';
+import { renderWithProvider } from '../../../../../test/lib/render-helpers';
+import configureStore from '../../../../store/store';
+import { GasFeeContextProvider } from '../../../../contexts/gasFee';
+
+import { GAS_ESTIMATE_TYPES } from '../../../../../shared/constants/gas';
+import BasefeeInput from './basefee-input';
+
+jest.mock('../../../../store/actions', () => ({
+  disconnectGasFeeEstimatePoller: jest.fn(),
+  getGasFeeEstimatesAndStartPolling: jest
+    .fn()
+    .mockImplementation(() => Promise.resolve()),
+  addPollingTokenToAppState: jest.fn(),
+}));
+
+const render = (txProps) => {
+  const store = configureStore({
+    metamask: {
+      ...mockState.metamask,
+      accounts: {
+        [mockState.metamask.selectedAddress]: {
+          address: mockState.metamask.selectedAddress,
+          balance: '0x1F4',
+        },
+      },
+      advancedGasFee: { maxBaseFee: 2 },
+      featureFlags: { advancedInlineGas: true },
+      gasFeeEstimates:
+        mockEstimates[GAS_ESTIMATE_TYPES.FEE_MARKET].gasFeeEstimates,
+    },
+  });
+
+  return renderWithProvider(
+    <GasFeeContextProvider
+      transaction={{
+        userFeeLevel: 'custom',
+        ...txProps,
+      }}
+    >
+      <BasefeeInput />
+    </GasFeeContextProvider>,
+    store,
+  );
+};
+
+describe('BasefeeInput', () => {
+  it('should renders advancedGasFee.baseFee value if current estimate used is not custom', () => {
+    render({
+      userFeeLevel: 'high',
+    });
+    expect(document.getElementsByTagName('input')[0]).toHaveValue(2);
```

Comment:
Thoughts on using `data-testid` instead of targeting an input directly like this?

Reply:
I avoid using `data-testid` I tried to use `getByRole` but somehow did not worked in this case.